.circleci/images/primary/Dockerfile-2.1.10

```diff
@@ -1,9 +1,92 @@
-# Last version: https://github.com/docker-library/ruby/blob/99def14400fcd612782367830836dfcbc10c8c50/2.1/Dockerfile
-FROM ruby:2.1.10
+# ruby:2.1.10 was based on jessie, but never got a aarch64 image
+# aarch64 used to be supported upon release but is not part of LTS, so archive.debian.org does not contain aarch64: https://www.debian.org/releases/jessie/
+
+# taken from https://github.com/docker-library/ruby/blob/c5693b25aa865489fee130e572a3f11bccebd21b/2.1/Dockerfile
+FROM buildpack-deps:stretch AS ruby-2.1.10-stretch
+
+# skip installing gem documentation
+RUN mkdir -p /usr/local/etc \
+  && { \
+    echo 'install: --no-document'; \
+    echo 'update: --no-document'; \
+  } >> /usr/local/etc/gemrc
+
+ENV RUBY_MAJOR 2.1
+ENV RUBY_VERSION 2.1.10
+ENV RUBY_DOWNLOAD_SHA256 5be9f8d5d29d252cd7f969ab7550e31bbb001feb4a83532301c0dd3b5006e148
+ENV RUBYGEMS_VERSION 2.6.12
+
+# some of ruby's build scripts are written in ruby
+#   we purge system ruby later to make sure our final image uses what we just built
+RUN set -ex \
+  \
+  && buildDeps=' \
+    bison \
+    dpkg-dev \
+    libgdbm-dev \
+    # ruby 2.3 on stretch can only support libssl1.0-dev (libssl dev from buildpack-deps is 1.1.x)
+    libssl1.0-dev \
+    ruby \
+  ' \
+  && apt-get update \
+  && apt-get install -y --no-install-recommends $buildDeps \
+  && rm -rf /var/lib/apt/lists/* \
+  \
+  && wget -O ruby.tar.xz "https://cache.ruby-lang.org/pub/ruby/${RUBY_MAJOR%-rc}/ruby-$RUBY_VERSION.tar.xz" \
+  && echo "$RUBY_DOWNLOAD_SHA256 *ruby.tar.xz" | sha256sum -c - \
+  \
+  && mkdir -p /usr/src/ruby \
+  && tar -xJf ruby.tar.xz -C /usr/src/ruby --strip-components=1 \
+  && rm ruby.tar.xz \
+  \
+  && cd /usr/src/ruby \
+  \
+# hack in "ENABLE_PATH_CHECK" disabling to suppress:
+#   warning: Insecure world writable dir
+  && { \
+    echo '#define ENABLE_PATH_CHECK 0'; \
+    echo; \
+    cat file.c; \
+  } > file.c.new \
+  && mv file.c.new file.c \
+  \
+  && autoconf \
+  && gnuArch="$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)" \
+  && ./configure \
+    --build="$gnuArch" \
+    --disable-install-doc \
+    --enable-shared \
+  && make -j "$(nproc)" \
+  && make install \
+  \
+  && apt-get purge -y --auto-remove $buildDeps \
+  && cd / \
+  && rm -r /usr/src/ruby \
+  \
+  && gem update --system "$RUBYGEMS_VERSION"
+
+ENV BUNDLER_VERSION 1.15.1
```

Comment:
These rubygems/bundler versions get overridden further down on the last lines of the dockerfile. Consider perhaps merging them? (Same note applies to other dockerfiles)

Reply:
That's on the table, as well as other cleanups/fixes/unification, but I wanted this PR to be as much of a drop-in replacement as possible.

Reply:
I agree with leaving messing around with specific versions of rubygems/bundler for other PRs. Nevertheless, I do think this one is a regression, since the previous `Dockerfile` did not have this confusion of mentioning two different sets of rubygems/bundler versions.

Can I convince you to at least leave a comment next to the `ENV BUNDLER_VERSION 1.15.1` and `ENV RUBYGEMS_VERSION 2.6.12` statements mentioning that these are actually not the versions that end up in the image, as they get overridden further down the `Dockerfile`?

Reply:
Yeah, I will add that in form of a TODO